R/fun_makefile_create.R

```diff
@@ -36,52 +36,69 @@ deepstate_create_makefile <-function(package,fun_name){
   rinside_lib <- system.file("lib", package="RInside")
 
   # CPPFLAGS : headers inclusion
-  compiler_cppflags <- paste0("-I", rcpp_include) 
-  compiler_cppflags <- paste(compiler_cppflags, paste0("-I", rcppdeepstate_include)) 
-  compiler_cppflags <- paste(compiler_cppflags, paste0("-I", rcpparmadillo_include)) 
-  compiler_cppflags <- paste(compiler_cppflags, paste0("-I", rinside_include)) 
-  compiler_cppflags <- paste(compiler_cppflags, paste0("-I", deepstate_build))
-  compiler_cppflags <- paste(compiler_cppflags, paste0("-I", deepstate_include)) 
-  compiler_cppflags <- paste(compiler_cppflags, paste0("-I", qs_include)) 
-  compiler_cppflags <- paste(compiler_cppflags, paste0("-I", "${R_INCLUDE}"))
-  write_to_file <- paste0(write_to_file, "CPPFLAGS=",compiler_cppflags, "\n")
+  cppflags <- c(rcpp_include, rcppdeepstate_include, rcpparmadillo_include,
+                rinside_include, deepstate_build, deepstate_include, qs_include,
+                "${R_INCLUDE}")
+
+  compiler_cppflags <- paste(cppflags, collapse=" -I ")
+  cppflags_write <- paste0("CPPFLAGS=-I",compiler_cppflags)
 
   # LDFLAGS : libs inclusion
-  compiler_ldflags <- paste0("-L", rinside_lib, " -Wl,-rpath=", rinside_lib) 
-  compiler_ldflags <- paste(compiler_ldflags, "-L${R_LIB} -Wl,-rpath=${R_LIB}")
-  compiler_ldflags <- paste(compiler_ldflags, paste0("-L", deepstate_build, " -Wl,-rpath=", deepstate_build))
-  write_to_file <- paste0(write_to_file, "LDFLAGS=", compiler_ldflags, "\n")
+  ldflag <- function(path) paste0("-L", path, " -Wl,-rpath=", path) 
+  ldflags <- c(ldflag(rinside_lib), ldflag("${R_LIB}"), ldflag(deepstate_build))
```

Comment:
this is better! nice cleanup. the intent is much more clear this way.

Reply:
Thanks!